lib/operations/command_v2.js

```diff
@@ -31,6 +36,32 @@ class CommandOperationV2 extends OperationBase {
     // TODO: consider making this a non-enumerable property
     this.server = server;
 
+    const options = this.options;
+    const serverWireVersion = maxWireVersion(server);
+    const inTransaction = this.session && this.session.inTransaction();
+
+    if (this.readConcern && commandSupportsReadConcern(cmd) && !inTransaction) {
+      Object.assign(cmd, { readConcern: this.readConcern });
+    }
+
+    if (serverWireVersion >= SUPPORTS_WRITE_CONCERN_AND_COLLATION) {
+      if (this.writeConcern && this.hasAspect(Aspect.WRITE_OPERATION)) {
+        Object.assign(cmd, { writeConcern: this.writeConcern });
+      }
+
+      if (options.collation && typeof options.collation === 'object') {
+        Object.assign(cmd, { collation: options.collation });
+      }
+    }
+
+    if (typeof options.maxTimeMS === 'number') {
```

Comment:
are `maxTimeMS` and `comment` both universal to all commands?

Reply:
yes, though they aren't quite documented everywhere. `maxTimeMS` is the primary target for a global client-side operation timeout.